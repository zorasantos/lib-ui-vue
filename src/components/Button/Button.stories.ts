import type { Meta, StoryObj } from "@storybook/vue3";

import Button from "./Button.vue";

const meta = {
  title: "Button",
  component: Button,
  tags: ["autodocs"],
  argTypes: {},
} satisfies Meta<typeof Button>;

export default meta;

type Story = StoryObj<typeof Button>;

export const Default: Story = {
  args: {
    label: "Button",
    loading: false,
    block: false,
  },
};
