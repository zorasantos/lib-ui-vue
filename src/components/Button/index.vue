<script setup lang="ts">
defineProps<{
  text: string;
  addStyle?: string;
  loading: boolean;
}>();
</script>
<template>
  <button
    :disabled="loading"
    :class="`w-full text-white ${loading ? 'bg-gray-200' : 'bg-blue-600'} ${
      !loading && 'hover:bg-blue-700'
    }  focus:ring-2 focus:cursor-auto focus:outline-none focus:ring-blue-300 font-medium rounded-lg text-sm px-5 py-2.5 text-center flex justify-center ${addStyle}`"
    v-bind="$attrs"
  >
    <span v-if="!loading">{{ text }}</span>
    <div
      v-if="loading"
      class="inline-block h-6 w-6 animate-spin rounded-full border-2 border-solid border-current border-r-transparent border-blue-600 align-[-0.125em] motion-reduce:animate-[spin_1.5s_linear_infinite]"
      role="status"
    ></div>
  </button>
</template>
