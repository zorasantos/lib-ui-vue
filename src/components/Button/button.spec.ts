import { mount } from "@vue/test-utils";
import Button from "./index.vue";
describe("Button", () => {
  test("render default button", () => {
    const wrapper = mount(Button, {
      props: {
        text: "Login",
        loading: false,
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.find("button").text()).toBe("Login");
    expect(wrapper.find("button").classes("bg-blue-600")).toBe(true);
  });

  test("renders the button with loading spinner when loading", () => {
    const wrapper = mount(Button, {
      props: {
        text: "Submit",
        loading: true,
      },
    });

    expect(wrapper.find("button span").exists()).toBe(false);
    expect(wrapper.find("button div").exists()).toBe(true);
    expect(wrapper.find("button").classes("bg-gray-200")).toBe(true);
  });

  test("renders the button when add style", () => {
    const wrapper = mount(Button, {
      props: {
        addStyle: "cursor-auto",
        text: "Submit",
        loading: false,
      },
    });

    expect(wrapper.find("button").text()).toBe("Submit");
    expect(wrapper.find("button").classes("bg-blue-600")).toBe(true);
    expect(wrapper.find("button").classes("cursor-auto")).toBe(true);
  });
});
