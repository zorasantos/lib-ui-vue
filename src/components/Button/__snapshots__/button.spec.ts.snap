// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button > render default button 1`] = `
"<button class=\\"w-full text-white bg-blue-600 hover:bg-blue-700  focus:ring-2 focus:cursor-auto focus:outline-none focus:ring-blue-300 font-medium rounded-lg text-sm px-5 py-2.5 text-center flex justify-center undefined\\"><span>Login</span>
  <!--v-if-->
</button>"
`;
